accounts/utils.py

```diff
@@ -0,0 +1,80 @@
+from datetime import timedelta
+
+import requests
+from django.utils import timezone
+
+from accounts.settings import accounts_settings
+
+
+def authenticated_request(user, method, url,
+                          params=None, data=None, headers=None, cookies=None, files=None,
+                          auth=None, timeout=None, allow_redirects=True, proxies=None,
+                          hooks=None, stream=None, verify=None, cert=None, json=None):
+    """
+    Helper method to make an authenticated request using the user's access token
+    NOTE be ABSOLUTELY sure you only make a request to Penn Labs products, otherwise
+    you will expose user's access tokens to the URL you provide and bad things will
+    happen
+    """
+
+    # Access token is expired. Try to refresh access token
+    if user.accesstoken.expires_at < timezone.now():
+        if not refresh_access_token(user):
+            return None  # Couldn't update access token
+
+    # Update Headers
+    headers = headers or {}
```

Comment:
very javascripty of you

Reply:
more pythonic would just be making the default for headers be an empty dictionary. do `headers=dict()` in the function signature.

Reply:
Both `requests` and `requests-oauthlib` use similar notation to this, I can't say I understand why, but I'd rather follow their lead.

Reply:
Probably because they migrated from an older version of python that didn't have default parameters? either way, pretty against this notation from a style standpoint. It's much more clear when you're reading a function signature what's required and what isn't with a default parameter.

Reply:
So after looking into it, I'm pretty sure the reason the standard is to use `None` instead of `{}` is because you can run into weird mutability issues since python evaluates the default arguments only once, then uses the same objects for the life of the program. That probably won't be an issue here, but this way still appears to be the standard. I'm seeing it in multiple projects.

Reply:
Fixed